repose-aggregator/components/filters/body-patcher-filter/src/main/scala/org/openrepose/filters/bodypatcher/BodyPatcherFilter.scala

```diff
@@ -65,19 +65,12 @@ class BodyPatcherFilter @Inject()(configurationService: ConfigurationService)
         val jsonPatches: List[String] = filterJsonPatches(requestPatches)
         if (jsonPatches.nonEmpty) {
           logger.debug("Applying json patches on the request")
-          val originalValue: JsValue = Try({
-            val inputStream: ServletInputStream = httpRequest.getInputStream
-            if (inputStream.available() > 0) {
-              PJson.parse(inputStream)
-            } else {
-              logger.trace("No request body found, creating empty json")
-              PJson.obj()
-            }
-          }).recover({
-            case jpe: JsonParseException =>
-              logger.trace("Bad Json body")
-              throw new RequestBodyUnparseableException("Couldn't parse the body as json", jpe)
-          }).get
+          val originalValue: JsValue = Try(PJson.parse(httpRequest.getInputStream))
+            .recover({
+              case jpe: JsonParseException =>
+                logger.trace("Bad Json body")
+                throw new RequestBodyUnparseableException("Couldn't parse the body as json", jpe)
```

Comment:
To be clear, I absolutely love including the original exception inside of a new exception.  I'm just wondering out loud how we convey the error to the client.